ios/PacketTunnel/PacketTunnelProvider.swift

```diff
@@ -238,6 +239,16 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
         }
     }
 
+    private func logIfDeviceHasSameIP(than addresses: [IPAddressRange]) {
+        let hasIPv4SameAddress = addresses.compactMap { $0.address as? IPv4Address }
+            .contains { $0 == ApplicationConfiguration.sameIPv4 }
+        let hasIPv6SameAddress = addresses.compactMap { $0.address as? IPv6Address }
+            .contains { $0 == ApplicationConfiguration.sameIPv6 }
+
+        let isNotUsingSameIP = (hasIPv4SameAddress || hasIPv6SameAddress) == false ? " NOT " : " "
+        providerLogger.debug("Same IP is\(isNotUsingSameIP)being used")
```

Comment:
On first sight this really looked like it would miss a space between "is" and "being" when Same IP is being used. I'm not going to nitpick about your codebase, but if I wrote it I would have `IsNotUsingSameIp` be empty when it is being used and otherwise "NOT " and adjust formatting string accordingly. :)